resources/assets/lib/beatmap-discussions/editor.tsx

```diff
@@ -464,7 +464,7 @@ export default class Editor extends React.Component<Props, State> {
           // clear invalid beatmapId references (for pasted embed content)
           const beatmap = typeof node.beatmapId === 'number' ? this.props.beatmaps[node.beatmapId] : undefined;
           if (beatmap == null || beatmap.deleted_at != null) {
-            Transforms.setNodes(editor, {beatmapId: null}, {at: path});
+            Transforms.setNodes(editor, {beatmapId: undefined}, {at: path});
```

Comment:
it looks like setting node prop to `null` will actually unset the prop (thus `undefined`). As there's no check here, the original `setNodes` will just keep looping as `node.beatmapId` will always be `undefined` and `undefined !== null`.

Not sure if it's slate thing or something here changed the behaviour.

Reply:
Probably this

https://github.com/ianstormtaylor/slate/blob/6a1376332bbd2567336c444c57c1e64fdf706feb/packages/slate/src/transforms/general.ts#L202

Reply:
https://github.com/ianstormtaylor/slate/blob/95f402c59414331b2eeca9a19bd2c73c0ab6cd6c/packages/slate/src/transforms/node.ts#L637
This would imply using `undefined` is better to avoid constantly triggering `set_node` since `node[k]` is going to be `undefined` instead of `null`.

There's also the cases where we have checks like `timestamp !== this.props.element.timestamp`

Reply:
yeah, although probably should just check if it needs updating at all in the first place (which I removed two weeks ago and caused this bug)